### PR TITLE
Feature/vtn 8313 edit transcript

### PR DESCRIPTION
**Pull request by AlexOVeritone**
Reusing engine results redux from VTN-8769 branch. +Some UI fixes.
This will go to sdo-integration branch when CR on VTN-8769 is done

### CODE REVIEW

File: packages/veritone-widgets/src/redux/modules/mediaDetails/index.js [hunk 1]
```diff
@@ -415,9 +420,7 @@ export default createReducer(defaultState, {
       ...state,
       [widgetId]: {
         ...state[widgetId],
-        selectedEngineCategory: {
```
Comment (ChrisPelletier) [inline packages/veritone-widgets/src/redux/modules/mediaDetails/index.js]: revert this back to de-structured payload

File: packages/veritone-widgets/src/widgets/MediaDetails/index.js [hunk 2]
```diff
@@ -622,18 +618,37 @@ class MediaDetailsWidget extends React.Component {
...
-    if (get(this.props.selectedEngineCategory, 'categoryType') === 'face') {
-      const selectedEngine = find(this.props.selectedEngineCategory.engines, {
-        id: this.props.selectedEngineId
-      });
-      return (
-        !selectedEngine.hasUserEdits || this.props.isDisplayingUserEditedOutput
-      );
+    const selectedCategoryType = get(this.props.selectedEngineCategory, 'categoryType');
+    const selectedEngine = find(this.props.selectedEngineCategory.engines, { id: this.props.selectedEngineId });
+    if (selectedCategoryType === 'face' && get(selectedEngine, 'hasUserEdits') && !this.props.isDisplayingUserEditedOutput) {
```
Comment (ChrisPelletier) [inline packages/veritone-widgets/src/widgets/MediaDetails/index.js]: maybe change this to something like this?

const editableCategoryTypes = ['face', 'transcript'];
// includes is from lodash
if (includes(editableCategoryTypes, selectedCategoryType) && get(selectedEngine, 'hasUserEdits') && !this.props.isDisplayingUserEditedOutput) {
    return false;
}

Then if we add a new editable category type just add it to the array.

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/SnippetSegment/index.js [hunk 3]
```diff
@@ -72,20 +73,20 @@ export default class SnippetSegment extends Component {
...
       startMediaPlayHeadMs,
       stopMediaPlayHeadMs
     } = this.props;
 
     const contentComponents = content.fragments.map((entry, index) => {
       const startTime = entry.startTimeMs;
       const stopTime = entry.stopTimeMs;
+      const fragmentKey = entry.guid ?
+        entry.guid :
+        `snippet-fragment-${startTime}-${stopTime}-${entry.value ? entry.value.substr(0, 32) : ''}`;
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/SnippetSegment/index.js]: would prefer to see `const value = entry.value || ''`

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js [hunk 4]
```diff
@@ -64,13 +66,17 @@ export default class TranscriptEngineOutput extends Component {
     mediaPlayerTimeMs: number,
     mediaPlayerTimeIntervalMs: number,
     outputNullState: node,
-    bulkEditEnabled: bool
+    bulkEditEnabled: bool,
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js]: it's alarming that there are no required props whatsoever, is this intentional?

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js [hunk 5]
```diff
@@ -84,6 +90,11 @@ export default class TranscriptEngineOutput extends Component {
     };
   }
 
+  handleUserEditChange = evt => {
+    this.props.onToggleUserEditedOutput &&
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js]: i've noticed people doing this often, I originally had the assumption that this function was optional, but i'm not so sure. If it's not optional then declare the prop as required and remove this validation as it isn't necessary.

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptContent/index.js [hunk 6]
```diff
@@ -172,21 +173,21 @@ export default class TranscriptContent extends Component {
...
               stopTimeMs: entry.stopTimeMs,
               value: selectedWord
-            });
+            };
+            if (entry.guid) {
+              snippet.guid = entry.guid;
+            }
 
+            //---Update Snippet Data---
+            snippetSentences = snippetSentences + selectedWord + ' ';
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptContent/index.js]: i'd like to see a lot of the logic in this function cleaned up, it's really messy right now. i.e

`
          //---Update Start & Stop Time---
          (snippetStartTime === undefined ||
            snippetStartTime > entry.startTimeMs) &&
            (snippetStartTime = entry.startTimeMs);
          (snippetStopTime === undefined ||
            snippetStopTime < entry.stopTimeMs) &&
            (snippetStopTime = entry.stopTimeMs);
`

what's wrong with `if/else`?

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js (hunk 4 above)
Comment (AlexOVeritone) [reply packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js]: Added isRequired for the nested shapes, but the upper level peoperties should not be required. Some of those mus bo in pair - if one set - then the other one should be set, but mostly component should show what was passed into it.

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/OverviewSegment/index.js [hunk 7]
```diff
@@ -50,12 +50,13 @@ export default class OverviewSegment extends Component {
       const startTime = fragmentData.startTimeMs;
       const stopTime = fragmentData.stopTimeMs;
       const value = fragmentData.value;
+      const fragmentKey = fragmentData.guid ?
+        fragmentData.guid :
+        `overview-fragment-${startTime}-${stopTime}-${value ? value.substr(0, 32) : ''}`;
```
[the remark this answers is not in the document]
Comment (AlexOVeritone) [reply packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/OverviewSegment/index.js]: done

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/SnippetSegment/index.js (hunk 3 above)
Comment (AlexOVeritone) [reply packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/SnippetSegment/index.js]: done

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js (hunk 5 above)
Comment (AlexOVeritone) [reply packages/veritone-react-common/src/components/TranscriptEngineOutput/index.js]: it is optional prop.

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptContent/index.js (hunk 6 above)
Comment (AlexOVeritone) [reply packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptContent/index.js]: Converted this to if then else.

File: packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/OverviewSegment/index.js [hunk 8]
```diff
@@ -46,18 +46,18 @@ export default class OverviewSegment extends Component {
...
 
     const readContents = [];
-    content.fragments.forEach(fragmentData => {
-      const startTime = fragmentData.startTimeMs;
-      const stopTime = fragmentData.stopTimeMs;
-      const value = fragmentData.value;
-      const fragmentKey = fragmentData.guid ?
-        fragmentData.guid :
-        `overview-fragment-${startTime}-${stopTime}-${value ? value.substr(0, 32) : ''}`;
+    content.fragments.forEach(entry => {
```
Comment (charliemantle) [inline packages/veritone-react-common/src/components/TranscriptEngineOutput/TranscriptSegment/OverviewSegment/index.js]: `const readContents = content.fragments.map(entry => {})`

**Comment by AlexOVeritone** [comment]
Will be creating a different PR to merge this into feature/VTN-8313-user-edit-engine-results to avoid rebasing.

**Pull request closed without merging**